typescript/src/redemption.ts

```diff
@@ -0,0 +1,168 @@
+// @ts-ignore
+import bcoin from "bcoin"
+// @ts-ignore
+import wif from "wif"
+import { BigNumber } from "ethers"
+import {
+  RawTransaction,
+  UnspentTransactionOutput,
+  Client as BitcoinClient,
+} from "./bitcoin"
+
+/**
+ * Contains information needed to fulfill a redemption request.
+ */
+export interface RedemptionRequest {
+  /**
+   * The address that is the recipient of the redeemed Bitcoins.
+   * The type of the address must be P2PKH, P2WPKH, P2SH or P2WSH.
+   */
+  address: string
+
+  /**
+   * The amount of Bitcoins in satoshis that is requested to be redeemed.
+   * The actual value of the output in the Bitcoin transaction will be decreased
+   * by the sum of the fee share and the treasury fee for this particular output.
+   */
+  amount: BigNumber
+
+  /**
+   * The amount of Bitcoins in satoshis that is subtracted from the amount in
+   * redemption request and used to pay for the transaction.
+   * The value should not be greater than the max fee in the Bridge on-chain
+   * contract at the time the redemption request was made.
+   * The sum of all output fee values makes the total fee of the redemption
+   * transaction.
+   */
+  feeShare: BigNumber
+
+  /**
+   * The amount of Bitcoins in satoshis that is subtracted from the amount in
+   * redemption request and used to pay the treasury fee.
+   * The value should be equal exactly the value of treasury fee in the Bridge
+   * on-chain contract at the time the redemption request was made.
+   */
+  treasuryFee: BigNumber
+}
+
+// TODO: Description
+export async function redeemDeposits(
+  bitcoinClient: BitcoinClient,
+  walletPrivateKey: string,
+  mainUtxo: UnspentTransactionOutput,
+  redemptionRequests: RedemptionRequest[],
+  witness: boolean
+): Promise<void> {
+  const rawTransaction = await bitcoinClient.getRawTransaction(
+    mainUtxo.transactionHash
+  )
+
+  const mainUtxoWithRaw: UnspentTransactionOutput & RawTransaction = {
+    ...mainUtxo,
+    transactionHex: rawTransaction.transactionHex,
+  }
+
+  const transaction = await createRedemptionTransaction(
+    walletPrivateKey,
+    mainUtxoWithRaw,
+    redemptionRequests,
+    witness
+  )
+
+  // Note that `broadcast` may fail silently (i.e. no error will be returned,
+  // even if the transaction is rejected by other nodes and does not enter the
+  // mempool, for example due to an UTXO being already spent).
+  await bitcoinClient.broadcast(transaction)
+}
+
+// TODO: Description
+export async function createRedemptionTransaction(
```

Comment:
Would be good to expose that function in the main `TBTC` interface as well

Reply:
This function is now [exposed in the TBTC interface](https://github.com/keep-network/tbtc-v2/blob/793b5df4b506d1016d155f4556182479230fd28f/typescript/src/index.ts#L213)